### PR TITLE
[libgo] Update to 3.1

### LINKED ISSUE #11077: [libgo] build failure
**Host Environment**
- OS: Windows 10
- Compiler: msvc2015

**To Reproduce**
`./vcpkg install libgo`

**Failure logs** 
```
PS F:\vcpkg> .\vcpkg.exe install libgo
Computing installation plan...
The following packages will be built and installed:
  * boost-algorithm[core]:x86-windows
  * boost-atomic[core]:x86-windows
  * boost-chrono[core]:x86-windows
  * boost-container[core]:x86-windows
  * boost-context[core]:x86-windows
  * boost-date-time[core]:x86-windows
  * boost-exception[core]:x86-windows
  * boost-lexical-cast[core]:x86-windows
  * boost-math[core]:x86-windows
  * boost-modular-build-helper[core]:x86-windows
  * boost-range[core]:x86-windows
  * boost-regex[core]:x86-windows
  * boost-system[core]:x86-windows
  * boost-thread[core]:x86-windows
  * boost-unordered[core]:x86-windows
    libgo[core]:x86-windows
Additional packages (*) will be modified to complete this operation.
Starting package 1/16: boost-modular-build-helper:x86-windows
Building package boost-modular-build-helper[core]:x86-windows...
A suitable version of git was not found (required v2.26.2). Downloading portable git v2.26.2...
Extracting git...
-- Performing post-build validation
-- Performing post-build validation done
Building package boost-modular-build-helper[core]:x86-windows... done
Installing package boost-modular-build-helper[core]:x86-windows...
Installing package boost-modular-build-helper[core]:x86-windows... done
Elapsed time for package boost-modular-build-helper:x86-windows: 11.18 s
Starting package 2/16: boost-atomic:x86-windows
Building package boost-atomic[core]:x86-windows...
-- Using cached F:/vcpkg/downloads/boostorg-atomic-boost-1.72.0.tar.gz
-- Using source at F:/vcpkg/buildtrees/boost-atomic/src/ost-1.72.0-d6fbf09978
-- Building x86-windows-rel
-- Building x86-windows-rel done
-- Building x86-windows-dbg
-- Building x86-windows-dbg done
-- Packaging x86-windows-rel
-- Packaging x86-windows-rel done
-- Packaging x86-windows-dbg
-- Packaging x86-windows-dbg done
-- Packaging headers
-- Packaging headers done
-- Using cached F:/vcpkg/downloads/boost_LICENSE_1_0.txt
-- Installing: F:/vcpkg/packages/boost-atomic_x86-windows/share/boost-atomic/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package boost-atomic[core]:x86-windows... done
Installing package boost-atomic[core]:x86-windows...
Installing package boost-atomic[core]:x86-windows... done
Elapsed time for package boost-atomic:x86-windows: 3.542 s
Starting package 3/16: boost-system:x86-windows
Building package boost-system[core]:x86-windows...
-- Using cached F:/vcpkg/downloads/boostorg-system-boost-1.72.0.tar.gz
-- Using source at F:/vcpkg/buildtrees/boost-system/src/ost-1.72.0-221555e8d1
-- Building x86-windows-rel
-- Building x86-windows-rel done
-- Building x86-windows-dbg
-- Building x86-windows-dbg done
-- Packaging x86-windows-rel
-- Packaging x86-windows-rel done
-- Packaging x86-windows-dbg
-- Packaging x86-windows-dbg done
-- Packaging headers
-- Packaging headers done
-- Using cached F:/vcpkg/downloads/boost_LICENSE_1_0.txt
-- Installing: F:/vcpkg/packages/boost-system_x86-windows/share/boost-system/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package boost-system[core]:x86-windows... done
Installing package boost-system[core]:x86-windows...
Installing package boost-system[core]:x86-windows... done
Elapsed time for package boost-system:x86-windows: 2.603 s
Starting package 4/16: boost-chrono:x86-windows
Building package boost-chrono[core]:x86-windows...
-- Using cached F:/vcpkg/downloads/boostorg-chrono-boost-1.72.0.tar.gz
-- Using source at F:/vcpkg/buildtrees/boost-chrono/src/ost-1.72.0-3bc8259e03
-- Building x86-windows-rel
-- Building x86-windows-rel done
-- Building x86-windows-dbg
-- Building x86-windows-dbg done
-- Packaging x86-windows-rel
-- Packaging x86-windows-rel done
-- Packaging x86-windows-dbg
-- Packaging x86-windows-dbg done
-- Packaging headers
-- Packaging headers done
-- Using cached F:/vcpkg/downloads/boost_LICENSE_1_0.txt
-- Installing: F:/vcpkg/packages/boost-chrono_x86-windows/share/boost-chrono/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package boost-chrono[core]:x86-windows... done
Installing package boost-chrono[core]:x86-windows...
Installing package boost-chrono[core]:x86-windows... done
Elapsed time for package boost-chrono:x86-windows: 4.409 s
Starting package 5/16: boost-container:x86-windows
Building package boost-container[core]:x86-windows...
-- Using cached F:/vcpkg/downloads/boostorg-container-boost-1.72.0.tar.gz
-- Using source at F:/vcpkg/buildtrees/boost-container/src/ost-1.72.0-aef6ec5ec0
-- Building x86-windows-rel
-- Building x86-windows-rel done
-- Building x86-windows-dbg
-- Building x86-windows-dbg done
-- Packaging x86-windows-rel
-- Packaging x86-windows-rel done
-- Packaging x86-windows-dbg
-- Packaging x86-windows-dbg done
-- Packaging headers
-- Packaging headers done
-- Using cached F:/vcpkg/downloads/boost_LICENSE_1_0.txt
-- Installing: F:/vcpkg/packages/boost-container_x86-windows/share/boost-container/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package boost-container[core]:x86-windows... done
Installing package boost-container[core]:x86-windows...
Installing package boost-container[core]:x86-windows... done
Elapsed time for package boost-container:x86-windows: 4.269 s
Starting package 6/16: boost-regex:x86-windows
Building package boost-regex[core]:x86-windows...
-- Using cached F:/vcpkg/downloads/boostorg-regex-boost-1.72.0.tar.gz
-- Using source at F:/vcpkg/buildtrees/boost-regex/src/ost-1.72.0-a233385537
-- Building x86-windows-rel
-- Building x86-windows-rel done
-- Building x86-windows-dbg
-- Building x86-windows-dbg done
-- Packaging x86-windows-rel
-- Packaging x86-windows-rel done
-- Packaging x86-windows-dbg
-- Packaging x86-windows-dbg done
-- Packaging headers
-- Packaging headers done
-- Using cached F:/vcpkg/downloads/boost_LICENSE_1_0.txt
-- Installing: F:/vcpkg/packages/boost-regex_x86-windows/share/boost-regex/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package boost-regex[core]:x86-windows... done
Installing package boost-regex[core]:x86-windows...
Installing package boost-regex[core]:x86-windows... done
Elapsed time for package boost-regex:x86-windows: 20.49 s
Starting package 7/16: boost-range:x86-windows
Building package boost-range[core]:x86-windows...
-- Using cached F:/vcpkg/downloads/boostorg-range-boost-1.72.0.tar.gz
-- Using source at F:/vcpkg/buildtrees/boost-range/src/ost-1.72.0-0826811d83
-- Packaging headers
-- Packaging headers done
-- Using cached F:/vcpkg/downloads/boost_LICENSE_1_0.txt
-- Installing: F:/vcpkg/packages/boost-range_x86-windows/share/boost-range/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package boost-range[core]:x86-windows... done
Installing package boost-range[core]:x86-windows...
Installing package boost-range[core]:x86-windows... done
Elapsed time for package boost-range:x86-windows: 682.4 ms
Starting package 8/16: boost-lexical-cast:x86-windows
Building package boost-lexical-cast[core]:x86-windows...
-- Using cached F:/vcpkg/downloads/boostorg-lexical_cast-boost-1.72.0.tar.gz
-- Using source at F:/vcpkg/buildtrees/boost-lexical-cast/src/ost-1.72.0-4c1cfadf14
-- Packaging headers
-- Packaging headers done
-- Using cached F:/vcpkg/downloads/boost_LICENSE_1_0.txt
-- Installing: F:/vcpkg/packages/boost-lexical-cast_x86-windows/share/boost-lexical-cast/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package boost-lexical-cast[core]:x86-windows... done
Installing package boost-lexical-cast[core]:x86-windows...
Installing package boost-lexical-cast[core]:x86-windows... done
Elapsed time for package boost-lexical-cast:x86-windows: 207.1 ms
Starting package 9/16: boost-math:x86-windows
Building package boost-math[core]:x86-windows...
-- Using cached F:/vcpkg/downloads/boostorg-math-boost-1.72.0.tar.gz
-- Using source at F:/vcpkg/buildtrees/boost-math/src/ost-1.72.0-2786b6df16
-- Building x86-windows-rel
-- Building x86-windows-rel done
-- Building x86-windows-dbg
-- Building x86-windows-dbg done
-- Packaging x86-windows-rel
-- Packaging x86-windows-rel done
-- Packaging x86-windows-dbg
-- Packaging x86-windows-dbg done
-- Packaging headers
-- Packaging headers done
-- Using cached F:/vcpkg/downloads/boost_LICENSE_1_0.txt
-- Installing: F:/vcpkg/packages/boost-math_x86-windows/share/boost-math/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package boost-math[core]:x86-windows... done
Installing package boost-math[core]:x86-windows...
Installing package boost-math[core]:x86-windows... done
Elapsed time for package boost-math:x86-windows: 41.14 s
Starting package 10/16: boost-exception:x86-windows
Building package boost-exception[core]:x86-windows...
-- Using cached F:/vcpkg/downloads/boostorg-exception-boost-1.72.0.tar.gz
-- Using source at F:/vcpkg/buildtrees/boost-exception/src/ost-1.72.0-e499a7f223
-- Building x86-windows-rel
-- Building x86-windows-rel done
-- Building x86-windows-dbg
-- Building x86-windows-dbg done
-- Packaging x86-windows-rel
-- Packaging x86-windows-rel done
-- Packaging x86-windows-dbg
-- Packaging x86-windows-dbg done
-- Packaging headers
-- Packaging headers done
-- Using cached F:/vcpkg/downloads/boost_LICENSE_1_0.txt
-- Installing: F:/vcpkg/packages/boost-exception_x86-windows/share/boost-exception/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package boost-exception[core]:x86-windows... done
Installing package boost-exception[core]:x86-windows...
Installing package boost-exception[core]:x86-windows... done
Elapsed time for package boost-exception:x86-windows: 1.636 s
Starting package 11/16: boost-unordered:x86-windows
Building package boost-unordered[core]:x86-windows...
-- Using cached F:/vcpkg/downloads/boostorg-unordered-boost-1.72.0.tar.gz
-- Using source at F:/vcpkg/buildtrees/boost-unordered/src/ost-1.72.0-a206fe956a
-- Packaging headers
-- Packaging headers done
-- Using cached F:/vcpkg/downloads/boost_LICENSE_1_0.txt
-- Installing: F:/vcpkg/packages/boost-unordered_x86-windows/share/boost-unordered/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package boost-unordered[core]:x86-windows... done
Installing package boost-unordered[core]:x86-windows...
Installing package boost-unordered[core]:x86-windows... done
Elapsed time for package boost-unordered:x86-windows: 257.2 ms
Starting package 12/16: boost-algorithm:x86-windows
Building package boost-algorithm[core]:x86-windows...
-- Using cached F:/vcpkg/downloads/boostorg-algorithm-boost-1.72.0.tar.gz
-- Using source at F:/vcpkg/buildtrees/boost-algorithm/src/ost-1.72.0-316cc45503
-- Packaging headers
-- Packaging headers done
-- Using cached F:/vcpkg/downloads/boost_LICENSE_1_0.txt
-- Installing: F:/vcpkg/packages/boost-algorithm_x86-windows/share/boost-algorithm/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package boost-algorithm[core]:x86-windows... done
Installing package boost-algorithm[core]:x86-windows...
Installing package boost-algorithm[core]:x86-windows... done
Elapsed time for package boost-algorithm:x86-windows: 647.7 ms
Starting package 13/16: boost-date-time:x86-windows
Building package boost-date-time[core]:x86-windows...
-- Using cached F:/vcpkg/downloads/boostorg-date_time-boost-1.72.0.tar.gz
-- Using source at F:/vcpkg/buildtrees/boost-date-time/src/ost-1.72.0-7f4a27a945
-- Building x86-windows-rel
-- Building x86-windows-rel done
-- Building x86-windows-dbg
-- Building x86-windows-dbg done
-- Packaging x86-windows-rel
-- Packaging x86-windows-rel done
-- Packaging x86-windows-dbg
-- Packaging x86-windows-dbg done
-- Packaging headers
-- Packaging headers done
-- Using cached F:/vcpkg/downloads/boost_LICENSE_1_0.txt
-- Installing: F:/vcpkg/packages/boost-date-time_x86-windows/share/boost-date-time/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package boost-date-time[core]:x86-windows... done
Installing package boost-date-time[core]:x86-windows...
Installing package boost-date-time[core]:x86-windows... done
Elapsed time for package boost-date-time:x86-windows: 6.555 s
Starting package 14/16: boost-thread:x86-windows
Building package boost-thread[core]:x86-windows...
-- Using cached F:/vcpkg/downloads/boostorg-thread-boost-1.72.0.tar.gz
-- Using source at F:/vcpkg/buildtrees/boost-thread/src/ost-1.72.0-af57bfeb4d
-- Building x86-windows-rel
-- Building x86-windows-rel done
-- Building x86-windows-dbg
-- Building x86-windows-dbg done
-- Packaging x86-windows-rel
-- Packaging x86-windows-rel done
-- Packaging x86-windows-dbg
-- Packaging x86-windows-dbg done
-- Packaging headers
-- Packaging headers done
-- Using cached F:/vcpkg/downloads/boost_LICENSE_1_0.txt
-- Installing: F:/vcpkg/packages/boost-thread_x86-windows/share/boost-thread/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package boost-thread[core]:x86-windows... done
Installing package boost-thread[core]:x86-windows...
Installing package boost-thread[core]:x86-windows... done
Elapsed time for package boost-thread:x86-windows: 7.529 s
Starting package 15/16: boost-context:x86-windows
Building package boost-context[core]:x86-windows...
-- Using cached F:/vcpkg/downloads/boostorg-context-boost-1.72.0.tar.gz
-- Using source at F:/vcpkg/buildtrees/boost-context/src/ost-1.72.0-3a4f7e8dfa
-- Building x86-windows-rel
-- Building x86-windows-rel done
-- Building x86-windows-dbg
-- Building x86-windows-dbg done
-- Packaging x86-windows-rel
-- Packaging x86-windows-rel done
-- Packaging x86-windows-dbg
-- Packaging x86-windows-dbg done
-- Packaging headers
-- Packaging headers done
-- Using cached F:/vcpkg/downloads/boost_LICENSE_1_0.txt
-- Installing: F:/vcpkg/packages/boost-context_x86-windows/share/boost-context/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package boost-context[core]:x86-windows... done
Installing package boost-context[core]:x86-windows...
Installing package boost-context[core]:x86-windows... done
Elapsed time for package boost-context:x86-windows: 5.505 s
Starting package 16/16: libgo:x86-windows
Building package libgo[core]:x86-windows...
-- Note: libgo only supports static library linkage. Building static library.
-- Using cached F:/vcpkg/downloads/yyzybb537-libgo-v2.8.tar.gz
-- Using source at F:/vcpkg/buildtrees/libgo/src/v2.8-43db49e519
-- Using cached F:/vcpkg/downloads/XBased-xhook-e18c450541892212ca4f11dc91fa269fabf9646f.tar.gz
-- Extracting source F:/vcpkg/downloads/XBased-xhook-e18c450541892212ca4f11dc91fa269fabf9646f.tar.gz
-- Using source at F:/vcpkg/buildtrees/libgo/src/9fabf9646f-2f330be28a
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
    Command failed: "D:/Program Files (x86)/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v
    Working Directory: F:/vcpkg/buildtrees/libgo/x86-windows-dbg
    See logs for more information:
      F:\vcpkg\buildtrees\libgo\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/libgo/portfile.cmake:39 (vcpkg_install_cmake)
  scripts/ports.cmake:90 (include)


Error: Building package libgo:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: libgo:x86-windows
  Vcpkg version: 2020.02.04-nohash

Additionally, attach any relevant sections from the log files above. 
```

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/4550576/install-x86-windows-dbg-out.log)

### PR DESCRIPTION
**Pull request by NancyLi1013**
`libgo `current version 2.8 is broken since `execution_context.hpp` doesn't exist in boost-context modular of boost 1.72.

So I update to the latest version 3.1 to keep it work.

Fixes #11077

Note: No feature needs to test.

### CODE REVIEW

**Comment by NancyLi1013** [comment]
The regressions on x64-uwp and arm-uwp are like this:

`4>C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(1781,5): error MSB8024: Using static version of the C++ runtime library is not supported. `
 
It seems that the current version 3.1 doesn't support uwp. I will try to disable it until the new version can support uwp.

**Review by JackBoosY — APPROVED** [review]

File: ports/libgo/portfile.cmake [hunk 1]
```diff
@@ -1,16 +1,13 @@
-include(vcpkg_common_functions)
-
+vcpkg_fail_port_install(ON_ARCH "arm" ON_ATRGET "uwp")
```
Comment (strega-nil) [inline ports/libgo/portfile.cmake:1]:
```suggestion
vcpkg_fail_port_install(ON_ARCH "arm" ON_TARGET "uwp")
```

File: ports/libgo/portfile.cmake [hunk 2]
```diff
@@ -28,36 +25,22 @@ file(RENAME ${XHOOK_SOURCE_PATH} ${SOURCE_PATH}/third_party/xhook)
...
 endif()
 
 # Handle copyright
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/libgo RENAME copyright)
-file(INSTALL ${CURRENT_PORT_DIR}/libgo-config.cmake DESTINATION ${CURRENT_PACKAGES_DIR}/share/libgo)
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(INSTALL ${CURRENT_PORT_DIR}/libgo-config.cmake DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
 
-file(COPY ${CMAKE_CURRENT_LIST_DIR}/usage DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
+file(COPY ${CMAKE_CURRENT_LIST_DIR}/usage DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
```
Comment (strega-nil) [inline ports/libgo/portfile.cmake:46]:
```suggestion
file(COPY ${CMAKE_CURRENT_LIST_DIR}/usage DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
```

**Review by strega-nil — APPROVED** [review]
Just some minor changes! Thanks @NancyLi1013, I'm going to merge after CI run.

**Comment by NancyLi1013** [comment]
Thanks for your further review and update. @strega-nil

**Pull request merged**